src/views/htmlcontent/src/js/components/app.component.ts

```diff
@@ -442,6 +461,58 @@ export class AppComponent implements OnInit, AfterViewChecked {
     }
 
     /**
+     * Perform copy and do other actions for context menu on the messages component
+     */
+    handleMessagesContextClick(event: {type: string, selectedRange: IRange}): void {
+        switch (event.type) {
+            case 'copySelection':
+                let selectedText = event.selectedRange.text();
+                this.executeCopy(selectedText);
+                break;
+            default:
+                break;
+        }
+    }
+
+    openMessagesContextMenu(event: any): void {
+        event.preventDefault();
+        let selectedRange: IRange = this.getSelectedRangeUnderMessages();
+        this.messagesContextMenu.show(event.clientX, event.clientY, selectedRange);
+    }
+
+    getSelectedRangeUnderMessages(): IRange {
+        let selectedRange: IRange = undefined;
+        let msgEl = this._el.nativeElement.querySelector('#messages');
+        if (msgEl) {
+            selectedRange = this.getSelectedRangeWithin(msgEl);
+        }
+        return selectedRange;
+    }
+
+    getSelectedRangeWithin(el): IRange {
+        let selectedRange = undefined;
+        let sel = rangy.getSelection();
+        let elRange = <IRange> rangy.createRange();
+        elRange.selectNodeContents(el);
+        if (sel.rangeCount) {
+            selectedRange = sel.getRangeAt(0).intersection(elRange);
+        }
+        elRange.detach();
+        return selectedRange;
+    }
+
+    // Copy text as text
+    executeCopy(text: string): void {
+        let input = document.createElement('textarea');
+        document.body.appendChild(input);
```

Comment:
Might also be a good idea to ensure that this is never visible. Something like `input.style.position = 'absolute'; input.style.bottom = '100%'` might work.